contrib/pyln-client/pyln/client/lightning.py

```diff
@@ -86,10 +86,14 @@ def __repr__(self) -> str:
         """
         return str(self.millisatoshis) + "msat"
 
-    def to_satoshi(self) -> Decimal:
+    def to_satoshi(self, whole=False) -> Decimal:
         """
-        Return a Decimal representing the number of satoshis.
+        Return a Decimal representing the number of satoshis;
+        rounded up to the nearest satoshi iff whole=True
```

Comment:
This is deeply weird?  We usually expect truncation.

More importantly, Decimal has entire routines dedicated to this: to_integral_value().  It supports all kinds of rounding.

Reply:
We discussed this in https://github.com/ElementsProject/lightning/pull/4295 and came to the conclusion that we need to expose the rounding behavior correctly.

In the end we punted on the full rounding functionality, and just added a function that rounds up in https://github.com/ElementsProject/lightning/pull/4295/commits/72550f42b092c12be5ceb9c2ee25440a6fe58ac7